integration/network/service_test.go

```diff
@@ -321,6 +321,49 @@ func noServices(client client.ServiceAPIClient) func(log poll.LogT) poll.Result
 	}
 }
 
+func TestServiceWithDatapathPortInit(t *testing.T) {
+	skip.If(t, testEnv.OSType == "windows")
+	skip.If(t, versions.LessThan(testEnv.DaemonAPIVersion(), "1.40"), "DataPathPort was added in API v1.40")
+	defer setupTest(t)()
+	var ops = []func(*daemon.Daemon){}
+	var datapathPort uint32 = 7777
+	ops = append(ops, daemon.WithSwarmDataPathPort(datapathPort))
+	d := swarm.NewSwarm(t, testEnv, ops...)
+
+	cli := d.NewClientT(t)
+	defer cli.Close()
+
+	// Create a overlay network
+	name := "saanvisthira" + t.Name()
+	network.CreateNoError(t, context.Background(), cli, name,
+		network.WithDriver("overlay"))
+
+	var instances uint64 = 1
+	serviceName := "TestService" + t.Name()
+	serviceID := swarm.CreateService(t, d,
+		swarm.ServiceWithReplicas(instances),
+		swarm.ServiceWithName(serviceName),
+		swarm.ServiceWithNetwork(name),
+	)
+
+	poll.WaitOn(t, serviceRunningCount(cli, serviceID, instances), swarm.ServicePoll)
+
+	_, _, err := cli.ServiceInspectWithRaw(context.Background(), serviceID, types.ServiceInspectOptions{})
+	assert.NilError(t, err)
+
+	err = cli.ServiceRemove(context.Background(), serviceID)
+	assert.NilError(t, err)
+	d.SwarmLeave(true)
+	d.Stop(t)
+
+	// Clean up , set it back to original one to make sure other tests don't fail
+	datapathPort = 4789
+	ops = append(ops, daemon.WithSwarmDataPathPort(datapathPort))
+	d = swarm.NewSwarm(t, testEnv, ops...)
```

Comment:
I'm not sure why this would be needed; `swarm.NewSwarm()` creates a _new_ daemon, so other tests should not be affected?

Oh..hm... perhaps in the "remote daemon" situation, where it doesn't create a new daemon; https://github.com/moby/moby/blob/d3cc071bb98669545d4e3043c9bd85879292b815/integration/internal/swarm/service.go#L50-L61

Reply:
when I wrote test cases for local scope default address pool feature( where we change default IP address ), few of the old test cases were failing. My initial assumption was each time we spawn new daemon ,  everything is fresh. Following  test  cases should not affect. But I was wrong. so I added clean up code( setting back to default value) after I complete my testing .  I continue the same practice in all new features that I am involved.

Reply:
If this has to be done for all tests, perhaps we should have a utility for it, and make it part of the `defer setupTest(t)()`

(not needed to do that in this PR)